src/main/java/hudson/plugins/ec2/util/AmazonEC2Factory.java

```diff
@@ -0,0 +1,12 @@
+package hudson.plugins.ec2.util;
+
+import java.net.URL;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.ec2.AmazonEC2;
+
+public abstract class AmazonEC2Factory {
+
+    public abstract AmazonEC2 connect(AWSCredentialsProvider credentialsProvider, URL ec2Endpoint);
```

Comment:
If we're really going to use Guice properly, we're going to want to make these values also injected. The AWSCredentialsProvider is one type, though you'll need to create an annotation to make the URL injectable (or maybe use `@Named`?).

Once you get to that level, this factory class is no longer needed and can be converted into the Guice module you added.

Reply:
Should be simple enough to implement, though I'm a bit unsure on how to work with `@Named` but I'm sure I'll figure it out.